src/Compilers/CSharp/Test/Syntax/Parsing/SwitchExpressionParsingTests.cs

```diff
@@ -0,0 +1,1089 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests;
+
+public class SwitchExpressionParsingTests : ParsingTests
+{
+    public SwitchExpressionParsingTests(ITestOutputHelper output) : base(output)
+    {
+    }
+
+    [Fact]
+    public void TestErrantCaseInSwitchExpression1()
+    {
+        UsingExpression("""
+            x switch
+            {
+                case 0 => 1,
+                case 1 => 2,
+            }
+            """,
+            // (2,2): error CS1041: Identifier expected; 'case' is a keyword
```

Comment:
I wish there were an easy way to say that we expect a pattern here. An identifier is allowed here but it's just one of many things which can be put here. It's not necessarily the thing we *expected*.

Reply:
Agreed. But we do this everywhere in the parser. Would be nice to remove that all and have the clean 'X construct expected ' for each construct. 

I can add a specific error here if you'd like :-)

Reply:
I would say either go with ERR_UnexpectedToken here, or something like:
> error: A switch expression arm does not begin with a 'case' keyword

Reply:
Works for me

Reply:
added dedicated error message.